### PR TITLE
Design refresh: Branch/Tag/Commit selector + Repository selector

### PR DESCRIPTION
**Pull request by umpox**
## Overview
- Update styling and colors on Branch/Tag/Commit selector
- Update to a controlled popover and add dismiss icon
- Update GitReferenceNodes to use `badge` styles across app

Design issue: https://github.com/sourcegraph/sourcegraph/issues/19927

### Branch/Tag/Commit Selector
image

## Repository selector
image

Designs: https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/Design-Refresh-Systemization-source-of-truth?node-id=954%3A1976

## Still to come (Other PRs)
- Search contexts selector
- Show more changes

### CODE REVIEW

File: client/web/src/components/ConnectionPopover.scss [hunk 1]
```diff
@@ -14,7 +25,7 @@ $popover-item-padding-v: 0.325rem;
         display: flex;
         flex-direction: column;
 
-        min-height: 27rem; // avoid jitter when only loading indicator is shown
+        height: 27rem;
```
Comment (umpox) [inline client/web/src/components/ConnectionPopover.scss:28]: I'm confirming that we want this to be a fixed height, and if so the exact value we should aim for. Will comment back!

File: client/web/src/components/FilteredConnection.scss [hunk 2]
```diff
@@ -26,7 +26,7 @@
         border-bottom-right-radius: 0;
     }
     &--compact &__loader {
-        padding: $popover-item-padding-v $popover-item-padding-h;
+        padding: var(--popover-item-padding-v) var(--popover-item-padding-h);
```
Comment (umpox) [inline client/web/src/components/FilteredConnection.scss:29]: As this is FilteredConnection, these changes may cause _some_ unrelated regressions (although everything initially seems fine as far as I can tell). These changes are scoped until the refresh so it should be OK to merge, we can pick up any further regressions as new PRs if we find any

File: client/web/src/repo/commits/GitCommitNode.tsx [hunk 3]
```diff
@@ -165,7 +165,7 @@ export const GitCommitNode: React.FunctionComponent<GitCommitNodeProps> = ({
                             <ContentCopyIcon className="icon-inline" />
                         </button>
                     </code>
-                    <div className="git-commit-node__sha-and-parents-parents">
+                    <div className="d-flex">
```
Comment (umpox) [inline client/web/src/repo/commits/GitCommitNode.tsx:168]: This class didn't exist (and is a little strangely named 😀), we just want `d-flex` here

**Review by 5h1rU — APPROVED** [review]

File: client/web/src/components/ConnectionPopover.scss [hunk 4]
```diff
@@ -9,6 +6,16 @@ $popover-item-padding-v: 0.325rem;
...
     max-width: 70vw;
 
+    --popover-item-padding-h: 0.75rem;
+    --popover-item-padding-v: 0.325rem;
+
+    .theme-redesign & {
+        --popover-item-padding-h: 1rem;
+        --popover-item-padding-v: 0.25rem;
+        background-color: var(--color-bg-1);
+        border-radius: 5px;
```
[the remark this answers is not in the document]
Comment (umpox) [reply client/web/src/components/ConnectionPopover.scss:16]: It is! Specified in the [designs](https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/Design-Refresh-Systemization-source-of-truth?node-id=954%3A1976). If we continue to deviate from 3px maybe we should consider renaming that variable and discussing with design if we have any consistency with these. Will raise in design review!

File: client/web/src/components/ConnectionPopover.scss [hunk 5]
```diff
@@ -9,6 +6,16 @@ $popover-item-padding-v: 0.325rem;
     min-width: 25rem;
     max-width: 70vw;
 
+    --popover-item-padding-h: 0.75rem;
+    --popover-item-padding-v: 0.325rem;
+
+    .theme-redesign & {
```
Comment (pdubroy) [inline client/web/src/components/ConnectionPopover.scss:12]: It seems like we are using two different ways of writing styles specific to the redesign. The pattern you're using here is:

```css
.awesome-widget {
    /* default styles */

    .theme-redesign & {
        /* redesign styles */
    }
}
```

and the other pattern we are using (e.g. [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@fa1ccbd7509720aea0bd204716df0d3770ca028e/-/blob/client/shared/src/commandPalette/CommandList.scss#L10)) is:

```css
.awesome-widget {
    /* default styles */
}

.theme-redesign .awesome-widget {
    /* redesign styles */
}
```

Should we standardize on one of these? It seems like the first one has the advantage of keeping the redesign styles as close as possible to the default styles. Thoughts @umpox @5h1rU @valerybugakov ?

**Review by pdubroy — COMMENTED** [review]
lgtm with one question

File: client/web/src/components/ConnectionPopover.scss (hunk 5 above)
Comment (umpox) [reply client/web/src/components/ConnectionPopover.scss:12]: I agree the first one is preferable, will make it a lot easier to refactor once we need to remove some dead code

**Pull request merged**